### DISCUSSION TITLE
Lock file disabled by default?

### DISCUSSION THREAD

**Discussion by x80486** [freshframework/fresh#2532] [General]
I see that `deno.jsonc` / `deno.json` by default has `"lock": false` for Fresh projects.

While I'm not sure if `deno.lock` fulfill the same guarantees that `package-lock.json` for Node.js projects or `go.sum` for Go projects, there must be good reasons for that file to exist.

On the other hand, I see that if I remove or change the `lock` flag, the file is generated correctly — but its content is updated not only on `deno task build`, but sometimes when I run the application and so on.

Shouldn't this be enabled by default? If so, what's the reasoning behind not using it.

**Comment by marvinhagemeister** [comment]
We originally had this enabled because `esm.sh` changed the way they transpiled their files a couple of times which made the hashes not match anymore. Plan is to re-enable the lockfile now that we're not depending anymore on `esm.sh` with the upcoming Fresh 2 release and using JSR instead.

I'll turn this into an issue so that we don't forget about this before doing the final release.

**Reply by x80486** [reply]
Got it. Perfect. Thanks! :raised_hands:
